src/client/lib/hooks/useInputValidityState.ts

```diff
@@ -44,28 +46,20 @@ export const useInputValidityState = (
 
   const onBlur = useCallback<React.FocusEventHandler<HTMLInputElement>>(
     (e) => {
-      // Transition error state when the input box loses focus.
-      transitionState(e.target.validity);
+      // Transition error state when the input box loses focus,
+      // but only if the input is not empty.
+      if (e.target.value !== '') {
+        transitionState(e.target.validity);
+      }
     },
     [transitionState],
   );
 
   const onInput = useCallback<React.FormEventHandler<HTMLInputElement>>(
     (e) => {
-      // We check the `composed` variable to see if the event originated from user input.
-      // This is so that we can run validation when users submit values through a password manager.
-      // Composed is not supported in IE11, so we check to see if it's undefined first.
-      const notOriginatingFromUserInput =
-        e.nativeEvent.composed !== undefined && !e.nativeEvent.composed;
-
-      // We also validate the field when filled via an autocomplete action in the browser.
-      // We can tell when this happens because the `data` property will be undefined.
-      // Usually from a keystroke, `data` would contain the value.
-      const originatingFromAutoComplete =
-        e.nativeEvent.composed !== undefined &&
-        (e.nativeEvent as Event & { data?: string })?.data === undefined;
-
-      if (notOriginatingFromUserInput || originatingFromAutoComplete) {
```

Comment:
this functionality wasn't actually working as expected, and actually caused issues with the validity not update on input, as both were always set to false, regardless of autocomplete or password manager.

So instead I removed this and replaced this with a check to see if the input was one of the specific "invalid" states and only transition the validity state if not.

Reply:
This makes sense! Simpler is better when we're trying to cover a bunch of browsers at once...